README.md

```diff
@@ -539,107 +619,131 @@ module.exports = {
-  resolveTestPath: (snapshotFilePath, snapshotExtension) =>
-    path.join(process.env.TEST_ROOT, path.basename(snapshotFilePath, snapshotExtension)),
-  testPathForConsistencyCheck: path.join(process.env.TEST_ROOT, 'example.test.js'),
+  async postRender(page, context) {
+    // Get entire context of a story, including parameters, args, argTypes, etc.
+    const storyContext = await getStoryContext(page, context);
+    if (storyContext.parameters.theme === 'dark') {
+      // do something
+    } else {
+      // do something else
+    }
+  },
 };
 ```
 
-### Image snapshot recipe
+It's useful for skipping or enhancing use cases like [image snapshot testing](#image-snapshot), [accessibility testing](#accessibility-testing) and more.
 
-Here's a slightly different recipe for image snapshot testing:
+#### waitForPageReady
+
+The `waitForPageReady` utility is useful when you're executing [image snapshot testing](#image-snapshot) with the test-runner. It encapsulates a few assertions to make sure the browser has finished downloading assets.
 
 ```js
 // .storybook/test-runner.js
 const { waitForPageReady } = require('@storybook/test-runner');
-const { toMatchImageSnapshot } = require('jest-image-snapshot');
-
-const customSnapshotsDir = `${process.cwd()}/__snapshots__`;
 
 module.exports = {
-  setup() {
-    expect.extend({ toMatchImageSnapshot });
-  },
   async postRender(page, context) {
     // use the test-runner utility to wait for fonts to load, etc.
     await waitForPageReady(page);
 
-    // If you want to take screenshot of multiple browsers, use
-    // page.context().browser().browserType().name() to get the browser name to prefix the file name
-    const image = await page.screenshot();
-    expect(image).toMatchImageSnapshot({
-      customSnapshotsDir,
-      customSnapshotIdentifier: context.id,
-    });
+    // by now, we know that the page is fully loaded
   },
 };
 ```
 
-There is also an exported `TestRunnerConfig` type available for TypeScript users.
-
-### Render lifecycle
+#### StorybookTestRunner user agent
 
-To visualize the test lifecycle, consider a simplified version of the test code automatically generated for each story in your Storybook:
+The test-runner adds a `StorybookTestRunner` entry to the browser's user agent. You can use it to determine if a story is rendering in the context of the test runner. This might be useful if you want to disable certain features in your stories when running in the test runner, though it's likely an edge case.
 
 ```js
-it('button--basic', async () => {
-  // filled in with data for the current story
-  const context = { id: 'button--basic', title: 'Button', name: 'Basic' };
+export const MyStory = () => {
+  const isTestRunner = window.navigator.userAgent.match(/StorybookTestRunner/);
+  return (
+    <div>
+      <p>Is this story running in the test runner?</p>
+      <p>{isTestRunner ? 'Yes' : 'No'}</p>
+    </div>
+  );
+};
+```
```

Comment:
@yannbf, one quick question, can't this be turned into a CSF 3 story example? I'm aware that this is an edge case, but I think porting it over can save us some time and questions on how to set it up with other frameworks (Vue, Angular, etc). If possible, we could simplify it using the `play` function for instance.

Reply:
Hey @jonniebigodes ! I'll gladly change to CSF3, but I don't think it's a good idea to show this as an example for the play function. It's mostly useful for people who have components that throw an error, and in the test-runner they want to render an empty div or something. I wouldn't want users to have a play function that behaves differently when running their tests 🤔

Reply:
Thanks for the clarification @yannbf. Appreciate it 🙏 ! Sounds reasonable, I just asked based on the text that's below this example and wanted to get a bit of input.